src/alpaka/AlpakaCore/getDevIndex.h

```diff
@@ -0,0 +1,17 @@
+#ifndef HeterogeneousCore_AlpakaUtilities_getDevIndex_h
+#define HeterogeneousCore_AlpakaUtilities_getDevIndex_h
+
+#include "AlpakaCore/alpakaConfigAcc.h"
+
+namespace cms::alpakatools {
+  template <typename Device>
+  inline int getDevIndex(Device const& dev) {
+    return 0;
+  }
+
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
```

Comment:
Sorry, I messed up with the comment, here is what I wanted to suggest:

```c++
#include <alpaka/alpaka.hpp>

namespace cms::alpakatools {

  // generic interface, for DevOacc and DevOmp5
  template <typename TDev>
  int getDevIndex(TDev const& device) {
    return device.iDevice();
  }

  // specialisation for DevCpu
  template <>
  int getDevIndex<alpaka::DevCpu>(alpaka::DevCpu const& device) {
    return 0;
  }

#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
  // specialisation for DevCudaRt
  template <>
  int getDevIndex<alpaka::DevCudaRt>(alpaka::DevCudaRt const& device) {
    return device.m_iDevice;
  }
#endif  // ALPAKA_ACC_GPU_CUDA_ENABLED

}  // namespace cms::alpakatools
```

That is, use `DevCudaRt` instead of `DevUniformCudaHipRt`, since we use `AccGpuCudaRt` etc.

Good point about `DevCpu` being always defined.